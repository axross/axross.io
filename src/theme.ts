import Theme from './core/Theme';

const theme: Theme = {
  spacing: {
    small: 8,
    regular: 16,
    large: 24,
    huge: 40,
    enormous: 80,
  },
  typography: {
    body1: {
      size: 16,
      weight: 400,
    },
    body2: {
      size: 14,
      weight: 400,
    },
    heading1: {
      size: 20,
      weight: 400,
    },
    heading2: {
      size: 24,
      weight: 400,
    },
    heading3: {
      size: 45,
      weight: 900,
    },
    heading4: {
      size: 56,
      weight: 900,
    },
    heading5: {
      size: 112,
      weight: 900,
    },
  },
};

export default theme;
