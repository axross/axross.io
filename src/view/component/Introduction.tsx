import { ClassAttributes, createElement, ReactNode } from 'react';
import Text from './Text';

type Props = ClassAttributes<HTMLElement> & {
  className?: string;
  children: ReactNode;
};

const Introduction = ({ className, children }: Props) => (
  <Text type="body1" sentence className={className}>
    {children}
  </Text>
);

export default Introduction;
