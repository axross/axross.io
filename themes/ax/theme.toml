# theme.toml template for a Hugo theme
# See https://github.com/spf13/hugoThemes#themetoml for an example

name = "Ax"
license = "MIT"
licenselink = "https://github.com/axross/axross.io/blob/master/LICENSE.md"
description = ""
homepage = "http://axross.io/"
tags = []
features = []
min_version = 0.19

[author]
  name = "axross"
  homepage = "http://axross.io"
