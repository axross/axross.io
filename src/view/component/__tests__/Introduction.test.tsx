import { ThemeProvider } from 'emotion-theming';
import { createElement } from 'react';
import { create } from 'react-test-renderer';
import theme from '../../../theme';
import Introduction from '../Introduction';

describe('<Introduction>', () => {
  test('it matches with the previous snapshot', () => {
    const renderer = create(
      <ThemeProvider theme={theme}>
        <Introduction className="berlin">text</Introduction>
      </ThemeProvider>,
    );

    expect(renderer.toJSON()).toMatchSnapshot();
  });
});
