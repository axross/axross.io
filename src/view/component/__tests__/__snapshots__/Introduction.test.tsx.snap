// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Introduction> it matches with the previous snapshot 1`] = `
.emotion-0 {
  display: block;
  line-height: 1.5;
  color: #fff;
  font-family: Montserrat;
  -webkit-font-smoothing: antialiased;
  font-size: 16px;
  font-weight: 400;
}

<span
  className="berlin emotion-0"
>
  text
</span>
`;
